src/us/kbase/workspace/kbase/KBaseWorkspaceConfig.java

```diff
@@ -583,4 +643,198 @@ public String getParamReport() {
 	public boolean hasErrors() {
 		return !errors.isEmpty();
 	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((adminReadOnlyRoles == null) ? 0 : adminReadOnlyRoles.hashCode());
+		result = prime * result + ((adminRoles == null) ? 0 : adminRoles.hashCode());
+		result = prime * result + ((auth2URL == null) ? 0 : auth2URL.hashCode());
+		result = prime * result + ((authURL == null) ? 0 : authURL.hashCode());
```

Comment:
this seems like another function with a lot of similar declarations that could be compacted down to a list to be iterated through.

Reply:
See notes for equals() above